util/src/main/java/io/zeebe/util/health/CriticalComponentsHealthMonitor.java

```diff
@@ -75,36 +75,55 @@ private void updateHealth() {
   }
 
   private void calculateHealth() {
-    final var status =
-        componentHealth.containsValue(HealthStatus.UNHEALTHY)
-            ? HealthStatus.UNHEALTHY
-            : HealthStatus.HEALTHY;
     final var previousStatus = healthStatus;
-    healthStatus = status;
-
-    if (previousStatus != status) {
-      switch (status) {
-        case HEALTHY:
-          if (failureListener != null) {
-            failureListener.onRecovered();
-          }
-          log.debug(
-              "The components are healthy. The current health status of components: {}",
-              componentHealth);
-          break;
-        case UNHEALTHY:
-          if (failureListener != null) {
-            failureListener.onFailure();
-          }
-          log.debug(
-              "Detected unhealthy components. The current health status of components: {}",
-              componentHealth);
-          break;
-        default:
-          log.warn("Unknown health status {}", status);
-          break;
-      }
+    healthStatus = calculateStatus();
+
+    if (previousStatus == healthStatus) {
+      return;
+    }
+
+    switch (healthStatus) {
+      case HEALTHY:
+        if (failureListener != null) {
+          failureListener.onRecovered();
+        }
+        break;
+
+      case UNHEALTHY:
+        if (failureListener != null) {
+          failureListener.onFailure();
+        }
+        break;
+
+      case DEAD:
```

Comment:
You should also add a test in `CriticalComponentsHealthMonitorTest` for this case.